### PR TITLE
Moves builFilesExist to startup-checks

### LINKED ISSUE #6493: Move `builtFilesExist` check into utils/startup-check.js
I think that the `builtFilesExist` check should be moved out of the server index / boot flow, and into the startup checks, which live in `server/utils/startup-check.js`. I can't see any reason why this needs to live in `core/server/index.js`, it depends on a single key from the config, but so do other startup checks.

It also makes sense to me to do this particular check quite early on - if the files aren't present there's not much point trying to do anything else :wink: So placing it after the `contentPath()` checks makes sense IMO.

This will also result in it not needing to be translated, which as a console warning I don't think it should be anyhow.

### PR DESCRIPTION
**Pull request by javorszky**
Fixes #6493
- moved function to startup checks
- flipped file system checks to sync versions
- exits on first sign of trouble

### CODE REVIEW

File: core/server/utils/startup-check.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ var packages = require('../../../package.json'),
     mode = process.env.NODE_ENV === undefined ? 'development' : process.env.NODE_ENV,
     appRoot = path.resolve(__dirname, '../../../'),
     configFilePath = process.env.GHOST_CONFIG || path.join(appRoot, 'config.js'),
+    config = require('../config'),
```
Comment (ErisDS) [inline core/server/utils/startup-check.js]: Due to how and where these are run, you can't include config here - it needs to be done in the same way as it is in the other checks, E.g. https://github.com/TryGhost/Ghost/pull/6499/files#diff-61698d6ade731ad65fbc2225c40fd34dR183

Comment (javorszky) [inline core/server/utils/startup-check.js]: Hm, that means I'd also have to duplicate the config option default value merge here as well. That was the reason I included it here.

**Comment by javorszky** [comment]
Refreshed, now with file-loading the config file.

**Comment by ErisDS** [comment]
Sorry @javorszky I should've taken the time to explain this properly on the issue.

The startup-check.js file is a special file. It gets included in two places:
- https://github.com/TryGhost/Ghost/blob/925f3930d6a8e577139e75a9052ff4d7ae7b6835/package.json#L21-L21
- https://github.com/TryGhost/Ghost/blob/06959661c05275d58cd9d632adbb54c12f1bbe03/index.js#L9-L9

The first of these occurs before `npm install` is run. That's why you can't include config, or any other dependency with the exception of 1) core node modules like fs, and 2) direct file requires like package.json.

The `builtFilesExist` check should be able to work the same way as the `contentPath` check - if the path is not in the `config.js` then use the default, and the default can just be hardcoded from the appRoot. In fact with this check, it's highly unlikely this has ever been overridden so it isn't really necessary to even go so far as to check the `config.js` file.

The need for duplicated logic is annoying, and at some point we'll refactor config so that the defaults live in their own includable file. For now, hardcoding duplicate stuff in startup-check.js is totally fine, but including dependencies is definitely not.

**Comment by javorszky** [comment]
@ErisDS the latest commit has those changes. Gets the config directly, or if no config exists, uses the default.

**Comment by ErisDS** [comment]
I see that :) However, you've also added lodash as a dependency, which won't have been installed yet...

**Comment by javorszky** [comment]
Hm... brb, fixing

**Comment by javorszky** [comment]
Huh, turns out the fix isn't actually more than the lodash version. Unlodashed.

File: core/server/utils/startup-check.js [hunk 2]
```diff
@@ -227,6 +229,49 @@ checks = {
...
+            fileNames = fileNames.map(function (file) {
+                return file.replace('.', '.min.');
+            });
+        }
+
+        function checkExist(fileName) {
+            try {
+                fs.statSync(fileName);
+            } catch (e) {
+                console.error('\x1B[31mERROR: Unable to open some of the built files!\033[0m');
```
Comment (ErisDS) [inline core/server/utils/startup-check.js]: Can we have this error message be the same as it was before? https://github.com/TryGhost/Ghost/blob/4ba0a745dfb8591e36f45002ba517e9da62bc7b8/core/server/translations/en.json#L189-L189

Also, these messages aren't used anywhere else so they can be removed from en.json

**Comment by javorszky** [comment]
Refreshed.

**Pull request merged**